File: readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <ctype.h>
#include <string.h>

int main(void)
{
    int letters = 0;
    int words = 0;
    int sentences = 0;
    int spaces = 0;
    int specialchar = 0;

    string text = get_string("Text: \n");

    for(int i = 0;i < strlen(text);i++)
    {
        if((int)toupper(text[i]) >= 65 && (int)toupper(text[i]) <= 90){
            letters++;
        }

        if((int)text[i] == 32)
        {
            spaces++;
            words = spaces + 1;
        }

        if((int)text[i] == 33 || (int)text[i] == 46 || (int)text[i] == 63)
        {
            sentences++;
        }
    }

    float L = ((float)letters /(float)words) * 100;
    float S = ((float)sentences / (float)words) * 100;

    float index = ((0.0588 * L) - (0.296 * S)) - 15.8;


    int inteiro = index * 10;
    if(inteiro % 10 >= 5)
    {
        inteiro = (inteiro / 10) + 1;
    } else
    {
        inteiro = inteiro / 10;
    }


    //printf("index: %f\n", index);
    //printf("inteiro: %d\n", inteiro);

    if(inteiro < 1)
    {
        printf("Before Grade 1\n");
    } else if(inteiro >= 1 && inteiro <= 16)
    {
        printf("Grade %d\n", (int)inteiro);
    } else
    {
        printf("Grade 16+\n");
    }










}

File: week_6/readability/readability.py
from cs50 import get_string

text = get_string("Text: ")
letters = 0
words = 1
sentences = 0

for i in range(len(text)):
    if (text[i].isalpha() == True):
        letters += 1
    if (text[i] == " "):
        words += 1
    if (text[i] == "." or text[i] == "!" or text[i] == "?"):
        sentences += 1

L = (float(letters) / float(words)) * 100
S = (float(sentences) / float(words)) * 100

CL = ((0.0588 * L) - (0.296 * S)) - 15.8

grade = int(CL * 10)

if (grade % 10 >= 5):
    grade = (grade / 10) + 1
else:
    grade = grade / 10

if (grade < 1):
    print("Before Grade 1")
elif (grade >= 1 and grade <= 16):
    print(f"Grade {grade}")
else:
    print("Grade 16+")
